### PR TITLE
Refactored logic to convert InternalMetadata To Grpc/HTTP Metadata and explicitly adding span context to internal metadata

### LINKED ISSUE #1710: [Regression] Propagate Span context in the input binding path to app
Fix the issue to propagate the span context information to the app in the input binding path.This is a regression.

> /area runtime

## What version of Dapr?
0.8.0

## Expected Behavior
Input binding path should pass the span context information to the app.

## Actual Behavior
Due to change in 0.8.0, which causes regression wherein Input binding path is not filling up the span context in the context.

### PR DESCRIPTION
**Pull request by shalabhms**
# Description

Refactored the code logic to just convert InternalMetadata To Grpc/HTTP Metadata and explicitly fill the span context to internal data. This refactoring will also fix the issue #1710

## Issue reference

Closes #1710 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/messaging/v1/util.go [hunk 1]
```diff
@@ -188,37 +208,85 @@ func reservedGRPCMetadataToDaprPrefixHeader(key string) string {
...
 			continue
 		}
 		setHeader(reservedGRPCMetadataToDaprPrefixHeader(keyName), listVal.Values[0])
 	}
+
+	// converts trace data in internalMD to HTTP headers
+	tracedataFromInternalMetadataToHTTPHeader(internalMD, setHeader)
+}
+
+func tracedataFromInternalMetadataToHTTPHeader(internalMD DaprInternalMetadata, setHeader func(string, string)) {
```
Comment (youngbupark) [inline pkg/messaging/v1/util.go:237]: Can you please add unit-test for this ?

File: pkg/messaging/v1/util.go [hunk 2]
```diff
@@ -188,37 +208,85 @@ func reservedGRPCMetadataToDaprPrefixHeader(key string) string {
...
+			setHeader(traceparentHeader, listVal.Values[0])
+		}
+		listVal = internalMD[tracestateHeader]
+		if listVal != nil && listVal.Values[0] != "" {
+			setHeader(tracestateHeader, listVal.Values[0])
+		}
+	}
+}
+
+// AddSpanContextToInternalMetadata adds spancontext to Dapr internal metadata
```
Comment (youngbupark) [inline pkg/messaging/v1/util.go:258]: Please add unit-test for this.

**Comment by shalabhms** [comment]
@youngbupark , this pr is not ready for review yet - please avoid reviewing further.

**Comment by amanbha** [comment]
@shalabhms can we close this if its not relevant anymore.

**Comment by shalabhms** [comment]
> @shalabhms can we close this if its not relevant anymore.

@amanbha , yes I was experimenting this change but found that it is good to be at single place. Also not much changes anticipated in this code block.

**Pull request closed without merging**